### PR TITLE
test(data-model): inner-structure conformance (trichotomy + expect() + markup + §314 collapse)

### PR DESCRIPTION
**Pull request by danfuzz**
## Summary

Final pass of the `packages/data-model` test reorganization arc (#3672 →
#3678 → #3679 → this one): bring the within-file test structure into
conformance with the project's testing-rules cluster. This is the **inner-
structure conformance** companion to #3679's outer file-org pass.

**Test code only** — no source / no `deno.json` changes; zero hot-path
touch.

## Important framing for reviewers

**This is NOT a byte-identity conservation pass like #3672 / #3678 / #3679.**
Restructuring `describe()`-nesting and rewording assertions IS the
*intended* work, so test bodies legitimately change shape and the diff is
**large and noisy by design**. The right frame for reviewing the noise:

**Conservation is coverage + semantics, not bytes:**

- **`it()`-count preserved end-to-end: 1122 `it()`s** (1124 base −
  2 documented same-class `instanceof` self-tautology deletions per Dan
  review comment, on `FabricEpochDays`/`FabricEpochNsec`); spine
  preserved at every commit otherwise.
- **Every assertion still asserts the same thing** — same actual/expected
  values, same strictness; nothing dropped, nothing added, nothing
  weakened.
- **The restructuring is the deliberate change**, not an accident worth
  flagging.

## Scope (4 conformance axes + an outer §314 collapse)

1. **Trichotomy §330-348** — constructor() / instance / static `describe()`
   layering, applied to 12 class test files (7 in `fabric-instances/` +
   `FabricBytes`/`FabricEpochDays`/`FabricEpochNsec`/`FabricHash` in
   `fabric-primitives/` + `SchemaAndHash`).
2. **Function-test shape §350-353** — applied to non-class test files.
3. **`expect()`-over-`assert*()` §355-end** — ~70 `assert*()` sites in
   `schema-utils.test.ts` converted to `expect()` style; +3 in
   `schema-hash.test.ts`; +`SchemaAndHash`.
4. **Markup + `it()` phrasing §273-279** — title markup conformance,
   "should…" rewording.

Plus a final commit that **collapses 4 multi-top `describe()` files to a
single top-level `describe()` per §314**.

## Commit stack (current tip `c0e3c6f24`)

Initial 5-commit conformance pass:

1. `ad5c1c9aa` — fabric-instances trichotomy (7 class files)
2. `2ace49aea` — fabric-primitives + SchemaAndHash trichotomy + 3
   assert→expect (5 files)
3. `d1522ba1b` — at-source-driven reshape (bucket-by-what-tests-assert;
   7 files)
4. `2deeba308` — assert-trio (schema-hash + schema-utils) + should→returns
   rewords (2 files)
5. `53a70a851` — collapse 4 multi-top describes to single-top per §314

Dan-review fixup round (5 commits applying Dan's inline review comments):

6. `8195e04a0` — move superclass-asserts-subclass tests from
   `ExplicitTagValue` to the concrete subclasses (Dan review comment #1)
7. `a813220d2` — delete same-class `instanceof` self-tautologies (Dan
   review comment #2; net −2 `it()`s, accounting for the 1124 → 1122 delta)
8. `e33504695` — rename `clone-{if-necessary,for-mutation}.test.ts` →
   `cloneIfNecessary.test.ts` / `cloneForMutation.test.ts` (Dan review)
9. `a23f3009d` — apply §273-274 `it()`-title rewords per Flux audit
10. `c0e3c6f24` — apply `.toBeInstanceOf()` matcher at concrete-class
    sites + `EmptyReconstructionContext` `it()`-rewords (Dan review
    comments #3/#4/#5)

**`.toBeInstanceOf()` narrowing call-out:** `@std/expect`'s
`.toBeInstanceOf(Y)` declares its parameter as `AnyConstructor`
(non-abstract), so TypeScript rejects abstract classes (TS2345). All
Fabric base types (`FabricInstance`, `FabricNativeWrapper`,
`FabricPrimitive`, `ExplicitTagValue`) are abstract, so 24 of the 26
candidate sites in the matcher sweep can't use it and **retain the
original `expect(x instanceof Y).toBe(true|false)` form**. The matcher
was applied at the 2 concrete-target sites (`FrozenMap instanceof Map`
and `FrozenSet instanceof Set`, both in `frozen-builtins.test.ts`). The
narrowing is documented in `c0e3c6f24`'s commit message.

## Sanctioned deltas worth call-out

- **2 surviving `assert()` calls in `schema-hash.test.ts`** are kept per
  rules §391-394: TS type narrowing in `callIntern`;
  `expect().toBeInstanceOf` is not a TS assertion function, so the surviving
  `assert()`s are required for the narrowing to hold. Flagged in the
  authoring commit message and in an in-source comment.
- **Reported test-step count rises** because adding `describe()` wrappers
  adds reported "steps" — this is purely a reporting artifact; the
  underlying `it()`-coverage stays at **1122** (post-tautology-deletion
  spine).

## Local gate at the tip

Cedar's local pre-push gate at the current tip `c0e3c6f24` (post
full Dan-review fixup round, all 5 inline comments addressed):
`check` / `fmt` / `lint` ✓; full data-model suite passes. CI on the PR
re-verifies on the GitHub side.

## Review checks

- **`it()`-count conservation**: **1122** cases on the merged tip (1124
  base − 2 deliberate tautology deletions per Dan review); spine
  preserved at every commit otherwise.
- **Assertion-semantics conservation**: every `assert*()`→`expect()`
  conversion preserves actual/expected/strictness; no assertion dropped,
  added, or weakened.
- **Scope discipline**: `test/` only — no `src/`, no `deno.json`, no other
  packages touched. (Verified at-source against base `b4fc982d4`.)

## Perf-check note

This PR is test-code-only with **zero hot-path change**. If `Performance
Check` flakes red (job-level wall-time outlier), it is the sanctioned
`NEW_PERF_BASELINE` false-positive case (same diagnostic as #3678: check
the step-level metric — if the test step itself is flat, the job-level is
CI-host scheduling noise, not a regression). Override applied per the
labs-perf mechanism in that case.

---

— `upstream-liaison-bolt` (Claude Opus 4.7 (1M context)), shepherding
work authored by `coder-cedar`

---
## Summary by cubic
Reorganized `packages/data-model` tests to match our conventions and reduce noise, and fixed a regression in `cloneIfNecessary.test.ts` that briefly introduced two top-level `describe()` blocks. Test code only; no `src/` changes.

- **Refactors**
  - Applied class-test trichotomy to `fabric-instances/*`, `fabric-primitives/*`, and `SchemaAndHash`; moved subclass/supertype identity checks into subclass tests.
  - Collapsed to a single top-level `describe()` in `value-hash.test.ts`, `cloneIfNecessary.test.ts`, `deep-freeze.test.ts`, and `EmptyReconstructionContext.test.ts`.
  - Converted `assert*()` to `expect()` in `schema-utils.test.ts` and `schema-hash.test.ts` (kept 2 `assert()` calls for required TypeScript narrowing).
  - Adopted `expect().toBeInstanceOf()` for concrete classes (e.g., `Map`, `Set`); kept `instanceof` checks for abstract types (`FabricInstance`, `FabricNativeWrapper`, `FabricPrimitive`, `ExplicitTagValue`).
  - Renamed test files to lowerCamelCase: `clone-if-necessary.test.ts` → `cloneIfNecessary.test.ts`, `clone-for-mutation.test.ts` → `cloneForMutation.test.ts` (bodies unchanged).
  - Reworded `it()` titles to third-person; deleted two same-class `instanceof` tautology tests in `FabricEpochDays.test.ts` and `FabricEpochNsec.test.ts` (total now 1122 `it()`s).
  - Removed duplicate `.toBeInstanceOf()` assertions in `frozen-builtins.test.ts`.
  - Unwrapped a redundant inner `describe()` in `cloneIfNecessary.test.ts`.

- **Bug Fixes**
  - Fixed the `cloneIfNecessary.test.ts` inner-describe unwrap that accidentally produced two top-level `describe()` blocks; restored single-top-level structure per §314 (no coverage or assertion changes; `it()` count unchanged at 1122).

<sup>Written for commit 2fcb9847f4ddbf3cf17c7c13da84761a67b30159. Summary will update on new commits. <a href="https://cubic.dev/pr/commontoolsinc/labs/pull/3681?utm_source=github">Review in cubic</a></sup>

### CODE REVIEW

File: packages/data-model/test/fabric-instances/ExplicitTagValue.test.ts [hunk 1]
```diff
@@ -5,27 +5,34 @@ import { ProblematicValue } from "../../src/fabric-instances/ProblematicValue.ts
 import { ExplicitTagValue } from "../../src/fabric-instances/ExplicitTagValue.ts";
 
 describe("ExplicitTagValue", () => {
-  it("UnknownValue is an instance of ExplicitTagValue", () => {
+  // `ExplicitTagValue` is abstract; these exercise it via its concrete
+  // subclasses. The subtype-relationship checks cross-cut the class rather
```
Comment (danfuzz) [inline packages/data-model/test/fabric-instances/ExplicitTagValue.test.ts]: This and the next test should be on the concrete classes not on this one. That is, it's reasonable for a subclass test to check what it's superclass is, but it's kinda weird for a superclass to assert stuff about particular subclasses.

File: packages/data-model/test/fabric-primitives/FabricEpochDays.test.ts [hunk 2]
```diff
@@ -9,45 +9,53 @@ import {
...
-
-  it("wraps negative values (pre-epoch)", () => {
-    const sd = new FabricEpochDays(-365n);
-    expect(sd.value).toBe(-365n);
-  });
-
-  it("is instanceof FabricEpochDays", () => {
+  // Pure type-identity / supertype checks: cross-cutting carve-out per the
+  // rule (don't fit a single member, aren't construction mechanics).
+  it("is an instance of `FabricEpochDays`", () => {
```
Comment (danfuzz) [inline packages/data-model/test/fabric-primitives/FabricEpochDays.test.ts]: This test is pretty much a tautology. No need to check if `(new Blah()) instanceof Blah`. (I know there's weird edge cases, but I don't think it's worth having a test for it.) Likewise equivalent tests for other classes.

File: packages/data-model/test/fabric-primitives/FabricEpochNsec.test.ts [hunk 3]
```diff
@@ -9,51 +9,59 @@ import {
...
-  it("is instanceof FabricEpochNsec", () => {
+  // Pure type-identity / supertype checks: cross-cutting carve-out per the
+  // rule (don't fit a single member, aren't construction mechanics).
+  it("is an instance of `FabricEpochNsec`", () => {
     const sn = new FabricEpochNsec(42n);
     expect(sn instanceof FabricEpochNsec).toBe(true);
   });
 
-  it("is instanceof FabricPrimitive", () => {
+  it("is an instance of `FabricPrimitive`", () => {
```
Comment (danfuzz) [inline packages/data-model/test/fabric-primitives/FabricEpochNsec.test.ts:14]: Just noticed a few cases of `expect(x instanceof Blah).toBe(true)`, but isn't there `expect(x).toBeInstanceOf(Blah)`? That'd be a better fit if so.

File: packages/data-model/test/EmptyReconstructionContext.test.ts [hunk 4]
```diff
@@ -5,71 +5,77 @@ import {
...
+    });
 
-  it("the two args are independent (frozen-intent context with a situation message)", () => {
-    const ctx = new EmptyReconstructionContext(false, "deep-clone path.");
-    expect(ctx.shouldDeepFreeze).toBe(false);
-    expect(() => ctx.getCell({ id: "of:bafyX", path: [], space: "did:key:z1" }))
-      .toThrow(
-        "Cannot reconstruct cell reference `of:bafyX`: deep-clone path.",
-      );
+    it("the two args are independent (frozen-intent context with a situation message)", () => {
```
Comment (danfuzz) [inline packages/data-model/test/EmptyReconstructionContext.test.ts]: Suggest framing `it("correctly accepts the two-argument form")`.

File: packages/data-model/test/EmptyReconstructionContext.test.ts [hunk 5]
```diff
@@ -5,71 +5,77 @@ import {
...
+        "Cannot reconstruct cell reference `of:bafyDEFAULT`: no runtime context provided.",
+      );
+    });
 
-  it("shouldDeepFreeze arg is forwarded (false stays false)", () => {
-    expect(new EmptyReconstructionContext(false).shouldDeepFreeze).toBe(false);
-    expect(new EmptyReconstructionContext(true).shouldDeepFreeze).toBe(true);
-  });
+    it("shouldDeepFreeze arg is forwarded (false stays false)", () => {
+      expect(new EmptyReconstructionContext(false).shouldDeepFreeze).toBe(
```
Comment (danfuzz) [inline packages/data-model/test/EmptyReconstructionContext.test.ts:57]: Suggest framing ```it("correctly passes `shouldDeepFreeze` to the superclass)```.
